src/app/service/users/users.service.ts

```diff
@@ -33,7 +33,9 @@ export class UsersService {
         next: data => {
           localStorage.setItem("id", data.id),
             localStorage.setItem("isLoggedIn", data.token)
-          this.getUser()
+          this.getUser().pipe(take(1)).subscribe(data => {
```

Comment:
No need for this here, you’re already calling the method in app component

Reply:
If I don't have it here then  user is changed only when page is refreshed and not when logging in. And on login the navigation bar is still showing previous user firstName and lastName until user refreshes the page